packages/dapp/src/hooks/useKeysManager.ts

```diff
@@ -0,0 +1,156 @@
+import type { VerificationMethodReference } from '@windingtree/org.json-schema/types/org.json';
+import type { GenericStateRecord } from '../store';
+import { useState, useCallback } from 'react';
+import { useAppReducer } from '../store/reducer';
+
+export const allowedKeysTypes: string[] = ['EcdsaSecp256k1VerificationKey2019'];
+
+export type KeyType =
+	typeof allowedKeysTypes[number];
+
+export enum RevocationReason {
+  "unspecified",
+  "keyCompromise",
+  "affiliationChanged",
+  "superseded",
+  "cessationOfOperation",
+  "certificateHold",
+  "privilegeWithdrawn"
+}
+
+export interface KeyRecord extends GenericStateRecord {
+	id: string; // Unique Id generated using `simpleUid()` 
```

Comment:
`id` - is already defined on `GenericStateRecord`, so you can omit its declaration

Reply:
:heavy_check_mark: